### PR TITLE
Code Review - Light Energy

### PR DESCRIPTION
**Pull request by stellarxo**

### CODE REVIEW

File: Assets/Scripts/Light/LightEnergy.cs [hunk 1]
```diff
@@ -1,17 +1,16 @@
 using UnityEngine;
 using System.Collections;
 
+/// <summary>
+/// ???????
```
Comment (stellarxo) [inline Assets/Scripts/Light/LightEnergy.cs]: Please fill this in or let me know what to put @johnny-la

File: Assets/Scripts/Light/LightEnergy.cs [hunk 2]
```diff
@@ -1,25 +1,36 @@
...
-
-    private bool debugInfiniteLight = false;
-
     public delegate void LightChangedHandler(float currentLight);
-    /** Called when the GameObject's amount of stored light changes */
+    
+    // Called when the GameObject's amount of stored light changes
     public event LightChangedHandler LightChanged = delegate {};
 
     public delegate void LightDepletedHandler();
```
Comment (stellarxo) [inline Assets/Scripts/Light/LightEnergy.cs]: do these have to be public

**Comment by stellarxo** [comment]
I can't, it breaks everything

File: Assets/Scripts/Light/LightEnergy.cs (hunk 2 above)
Comment (stellarxo) [inline Assets/Scripts/Light/LightEnergy.cs]: Used by light source and light colour

File: Assets/Scripts/Light/LightEnergy.cs [hunk 3]
```diff
@@ -29,19 +38,19 @@ public LightEnergy(float defaultEnergy, GameObject gameObject, bool debugInfinit
     }
 
     /// <summary>
-    /// Adds the given amount of light energy
+    /// Adds the specified amount of light energy and
+    /// informs all all interested subscribers that the amount of energy
+    /// possessed by the game object has changed
     /// </summary>
     public void Add(float lightEnergy)
```
Comment (stellarxo) [inline Assets/Scripts/Light/LightEnergy.cs]: used by vortex and light source

**Comment by johnny-la** [comment]
I added the class comment @stellarxo

File: Assets/Scripts/Light/LightEnergy.cs (hunk 3 above)
Comment (alisterdev) [inline Assets/Scripts/Light/LightEnergy.cs]: @stellarxo The point is to extend mono behaviour only when absolutely needed

**Pull request merged**